src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Com.cs

```diff
@@ -12,6 +12,9 @@ namespace System.Runtime.InteropServices
 {
     public static partial class Marshal
     {
+        private static readonly Guid IID_IDispatch = new Guid("00020400-0000-0000-C000-000000000046");
```

Comment:
This will be part of the Marshal static constructor and thus this will run and the code (will all dependencies) be included in the binary even when the app is using Marshal, but not using  `GetIDispatchForObject` at all. I would instead inline the guid into GetIDispatchForObject

`return ComWrappers.ComInterfaceForObject(o, new Guid(0x00020400, ...) /* IID_IDispatch */);